### PR TITLE
Add Kuzzle to comparison table (fixes #294)

### LINKED ISSUE #294: Add kuzzle to the other solutions comparison matrix
In the [comparison matrix with other solutions](http://kinto.readthedocs.org/en/latest/overview.html#comparison-with-other-solutions), we should probably add an entry for http://kuzzle.io

As far as I could see it has no authentication/permission yet. Documentation does not say much about pluggability either.

### PR DESCRIPTION
**Pull request by leplatrem**
capture d ecran de 2015-11-23 12-37-31

* [x] Split core features from ecosystem
* [x] Update current features/roadmap
* [x] Clean gray background in revision illustration

capture d ecran de 2015-11-23 12-43-03

* [x] Mention kuzzle in comparison table (fixes #294) (/ping @scottinet)

### CODE REVIEW

**Comment by leplatrem** [comment]
@n1k0 r?

File: docs/overview.rst [hunk 1]
```diff
@@ -116,31 +165,34 @@ permission settings.
...
+File storage                 ✔ [#]_   ✔                  ✔        ✔
+Batch/bulk operations        ✔       ✔                  ✔
+Changes stream               ✔ [#]_   ✔       ✔         ✔                        ✔
+Pluggable authentication     ✔                          ✔        [#]_
+Pluggable storage / cache    ✔                                    ✔              ✔
+Self-hostable                ✔                          ✔        ✔              ✔
+Decentralised discovery      [#]_                                 ✔
+Open source                  ✔                          ✔        ✔              ✔
+Language                     Python                    Erlang   Node.js [#]_    Node.js
+===========================  ======  ======  ========  =======  ==============  =======
```
Comment (n1k0) [inline docs/overview.rst]: I'd probably list live notifications support for all solutions, mentioning the feature is upcoming through an upcoming plugin for kinto.

Comment (n1k0) [inline docs/overview.rst]: Ah, that's part of "Changes stream"; maybe you could rename it to "Real-time updates"?

**Comment by n1k0** [comment]
Awesome. r+wc

**Comment by scottinet** [comment]
Hi, and thank you for including Kuzzle in your comparison table!

About Kuzzle, as you may know, we're still in alpha version, and we are working hard to bring our product to beta. Nevertheless, we've already developped some of the features you mention, and I'd like to bring some precisions. :-)

But first, can you help me understand these two listed features, to be sure to answer correctly about these: ``Offline-first client`` and ``Decentralised discovery``?

About the table itself:

**Functionalities we already support:**

* Conflict resolution
* Bulk operations ([API documentation](https://github.com/kuzzleio/kuzzle/blob/master/docs/API.WebSocket.md#performing-a-bulk-import-on-a-data-collection)). Easy access through our SDK is planned in the next few weeks
* Validation, through [pipe plugins](https://github.com/kuzzleio/kuzzle/blob/master/docs/plugins.md#pipes)

**Functionalities in alpha stage:**

* Fine-grained permissions ([pull request](https://github.com/kuzzleio/kuzzle/pull/63)). We already have a [dev documentation](https://github.com/kuzzleio/kuzzle/tree/master/lib/api/core/models/repositories), and default roles are available in our [configuration file](https://github.com/kuzzleio/kuzzle/blob/master/.kuzzlerc). More documentation and features are currently under development

**Under development:**

* Pluggable authentication

**Comment by leplatrem** [comment]
Thank your @scottinet for the feedback !

> Offline-first client and Decentralised discovery?

The offline-first client allows application developers to implement an application that works offline. The developers would just create/update/delete records, and the library takes care of synchronizing the local database with the remote server. See [kinto.js](https://github.com/Kinto/kinto.js) or [PouchDB](http://pouchdb.com/)

Decentralized discovery is the ability for the application to store the data at a specific location automatically. Most likely that depends on the authenticated user. When I use an application and say that I am mathieu@mydomain.com then the application knows that my Kinto instance is hosted at https://kinto.framasoft.net. See [remotestorage](https://remotestorage.io/) for example.

> Conflict resolution

Do you have any details to provide ? This is a very wide topic and if you do something in particular I think it would be great to mention it in our docs :) 
I realize that *Kinto* does not really perform any automatic conflict resolution. It's more about *concurrency control*, using `Etag`.

> Bulk operations 

Noted. From what I read, Kuzzle does only bulk create (import) though.

> Validation, through pipe plugins

Noted ! 
I think we should probably distinguish what is available via code, and via HTTP API. In for example Kinto you defined a JSON Schema on the collection via the HTTP API, and created/updated records will be validated for this collection.

> Fine-grained permissions

Great!
I read the documentation (and almost the whole pull-request) and from what I understood (may be wrong), the permissions are set via configuration (or code), but not via HTTP API.

> Pluggable authentication

Noted :)


Again, thanks a lot for your feedback!

> We couldn't manage to come to #OSSPARIS15, but I'm sure we'll be able to meet soon and exchange about our respective visions and roadmaps!

**Comment by scottinet** [comment]
Hi again :)

* Offline-first: do you mean [this](http://kuzzleio.github.io/sdk-documentation/#offline-mode)? :-)
* Decentralized discovery: we don't support data zones yet
* Conflict resolution: we simply expose [ElasticSearch Conflict Management](https://www.elastic.co/guide/en/elasticsearch/guide/current/version-control.html). Usually, it's a "last update wins" strategy, but several options are available to fine-tune this. For instance, updates on critical documents can be done using the '_version' tag, telling ElasticSearch that we want to update a specific version of the document. If the current document version is greater than the provided one, a conflict error occurs.
* Bulk: indeed, we support only bulk insert/update/delete for now. Data dumps functionalities are planned for a later time
* Validation: it seems that you're referring to data *format* validation, through a schema. We call them ``mappings`` and we [support them](https://github.com/kuzzleio/kuzzle/blob/master/docs/API.WebSocket.md#setting-up-a-data-mapping-in-a-collection) (and also in our [SDK](http://kuzzleio.github.io/sdk-documentation/#kuzzledatamapping)). I thought you were talking about functional validation, which is business-specific. For instance, it's very common for a back-end to allow data to be created by an application only if this data complies to certain conditions. That's why I mentioned plugins
* Fine-grained permissions: like I said, this functionality is still in alpha-stage. For instance, a new API route has just been submitted in this [pull request](https://github.com/kuzzleio/kuzzle/pull/83), allowing to create or update roles through the API.

**Comment by leplatrem** [comment]
Great, thank you!

I hadn't seen the SDK documentation yet. Neat! 

It looks like Kuzzle and Kinto columns will be pretty similar then :)

**Comment by tarekziade** [comment]
There's one thing I find unclear in the comparison matrix: some of the contenders are client-server products, and some are just servers.  For instance I don't think we're just comparing Kinto with the Firebase *Server* only, but rather with the Firebase **product**.

In the same vein, comparing Kinto with CouchDB might not be fair to CouchDB if we don't include PouchDB in the matrix.

I think we should treat everyone equally by comparing **products** - and before the matrix, clearly define what is the product we are talking about.

For example: "an offline-first toolkit" ? and for Kinto, the Kinto+Kinto.js combo ? for CouchDB the PouchDB+CouchDB combo ? etc.

**Comment by Natim** [comment]
> I think we should treat everyone equally by comparing products

:+1:

**Comment by leplatrem** [comment]
yes!
I will give it a try. Thanks for your feedback!

**Pull request merged**